policy/approval/approve.go

```diff
@@ -189,6 +190,11 @@ func (r *Rule) IsApproved(ctx context.Context, prctx pull.Context) (bool, string
 	// "author" is the user who opened the PR
 	// if contributors are allowed, the author counts as a contributor
 	author := prctx.Author()
+
+	if !r.Options.AllowAuthor && !r.Options.AllowNonAuthorContributor && !r.Options.AllowContributor {
```

Comment:
I don't think this condition is the correct one. If I'm reading this right, this combines with line 198 to reduce to something like:

```go
if (a && b && c) || (a && b) {
  banned[author] = true
}
```

That `if` condition reduces further to `a && b`, which is what the old code checks.

Instead of changing when the author is banned, I think we need to change when contributors are banned based on the setting of the flag (so some modification to the block starting on line 203.)

Reply:
ahh gotcha, yea this was hard to grok, will get this fixed up.